spec/acceptance/lint_spec.rb

```diff
@@ -62,4 +62,110 @@
       FactoryGirl.lint only_valid_factories
     end.not_to raise_error
   end
+
+  it 'allows for explicit factories and additional options' do
+    define_model 'User', name: :string do
+      validates :name, presence: true
+    end
+
+    FactoryGirl.define do
+      factory :user do
+        name 'assigned'
+      end
+    end
+
+    FactoryGirl.lint FactoryGirl.factories, validate_traits: true
+  end
+
+  it 'allows for additional options without explicit factories' do
+    define_model 'User', name: :string do
+      validates :name, presence: true
+    end
+
+    FactoryGirl.define do
+      factory :user do
+        name 'assigned'
+      end
+    end
+
+    FactoryGirl.lint validate_traits: true
+  end
+
+  describe 'trait validation' do
+
+    context 'turned on' do
+
+      it 'raises if a trait produces an invalid object' do
+        define_model 'User', name: :string do
+          validates :name, presence: true
+        end
+
+        FactoryGirl.define do
+          factory :user do
+            name 'Yep'
+            trait :unnamed do
+              name nil
+            end
+          end
+        end
+
+        error_message = <<-ERROR_MESSAGE.strip
+The following factories are invalid:
+
+* user/unnamed - Validation failed: Name can't be blank (ActiveRecord::RecordInvalid)
+        ERROR_MESSAGE
+
+        expect do
+          FactoryGirl.lint traits: true
+        end.to raise_error FactoryGirl::InvalidFactoryError, error_message
+
+      end
+
+      it 'does not raise if a trait produces a valid object' do
+        define_model 'User', name: :string do
+          validates :name, presence: true
+        end
+
+        FactoryGirl.define do
+          factory :user do
+            name 'Yep'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.